nexus/src/external_api/shared.rs

```diff
@@ -0,0 +1,221 @@
+    )]
+    #[display(style = "kebab-case")]
+    pub enum DummyRoles {
+        Bogus,
+    }
+
+    #[test]
+    fn test_policy_parsing() {
+        // Success case (edge case: max number of role assignments)
+        let role_assignment = serde_json::json!({
+            "identity_type": "user_builtin",
+            "identity_id": "75ec4a39-67cf-4549-9e74-44b92947c37c",
+            "role_name": "bogus"
+        });
+        const MAX: usize = MAX_ROLE_ASSIGNMENTS_PER_RESOURCE;
+        let okay_input =
+            serde_json::Value::Array(vec![role_assignment.clone(); MAX]);
+        let policy: Policy<DummyRoles> =
+            serde_json::from_value(serde_json::json!({
+                "role_assignments": okay_input
+            }))
+            .expect("unexpectedly failed with okay input");
+        assert_eq!(policy.role_assignments[0].role_name, DummyRoles::Bogus);
+
+        // Failure case: too many role assignments
+        let bad_input =
+            serde_json::Value::Array(vec![role_assignment; MAX + 1]);
+        let error =
+            serde_json::from_value::<Policy<DummyRoles>>(serde_json::json!({
+                "role_assignments": bad_input
+            }))
+            .expect_err("unexpectedly succeeded with too many items");
+        assert_eq!(
+            error.to_string(),
+            "invalid length 65, expected a list of at most 64 role assignments"
+        );
+    }
+
+    #[test]
+    fn test_role_assignment_from_database() {
+        let identity_id =
+            "75ec4a39-67cf-4549-9e74-44b92947c37c".parse().unwrap();
+        let resource_id =
+            "9e3e3be8-4051-4ddb-92fa-32cc5294f066".parse().unwrap();
+        let bad_input = db::model::RoleAssignment {
+            identity_type: db::model::IdentityType::UserBuiltin,
+            identity_id,
+            resource_type: ResourceType::Organization.to_string(),
+            resource_id,
+            role_name: String::from("bogosity"),
+        };
+
+        let ok_input = db::model::RoleAssignment {
+            role_name: String::from("bogus"),
+            ..bad_input.clone()
+        };
+
+        let error = <shared::RoleAssignment<DummyRoles>>::try_from(bad_input)
+            .expect_err("unexpectedly succeeding parsing database role");
+        println!("error: {:#}", error);
+        if let Error::InternalError { internal_message } = error {
+            assert_eq!(
+                internal_message,
+                "parsing database role assignment: unrecognized role name \
+                \"bogosity\": parse failed."
+            );
+        } else {
+            panic!(
+                "expected internal error for database parse failure, \
+                found {:?}",
+                error
+            );
+        }
+
+        let success = <shared::RoleAssignment<DummyRoles>>::try_from(ok_input)
+            .expect("parsing valid role assignment from database");
+        println!("success: {:?}", success);
```

Comment:
seems like useful to see if the test fails?

Reply:
I actually kept this on purpose so that if you were debugging this test you could see the output.  It's hidden unless you use `--nocapture` or the test fails.